### DISCUSSION TITLE
Custom Layouts

### DISCUSSION THREAD

**Discussion by kylemclaren** [scalar/scalar#908] [Q&A] [answered]
Thanks for this great project! So I know there is the modern layout and the classic layout: https://github.com/scalar/scalar/tree/6da73708130065bb75d17dbb4d04b9cb351123b0/packages/api-reference/src/components/Layouts

What would be the easiest way to create a custom layout (for instance, I want to use the header from classic, with the docs layout of modern) without maintaining my own fork of the repo?

**Comment by hanspagel** [comment]
Good question! That use case is not really intended yet. Need to think about that.

What’s it that you like about the classic header that you miss in the header of the modern layout?

**Reply by kylemclaren** [reply]
As far as I can tell, the modern layout does not have a header (unless on mobile) -- is there a way to enable it?

**Comment by kylemclaren** [comment]
I guess I can describe in more detail what I am looking to do and perhaps these can be extrapolated into their own issues:

All of the below assumes `modern` layout
- Toggle sidebar navigation (i.e. show/hide on demand in the same way as when on mobile)
- Ability to move dark mode toggle and search bar to header (like with `classic` layout)
- Ability to hide [scalar promo text](https://github.com/scalar/scalar/blob/6da73708130065bb75d17dbb4d04b9cb351123b0/packages/api-reference/src/components/DarkModeToggle.vue#L26)
- Ability to add a custom logo to header (bonus points if you can set light and dark mode logos)

**Reply by hwkr** [reply, marked as answer]
I think right now you can do all this by using the `ApiReferenceBase` component in a Vue app and then creating a layout wrapper similar to the [`ModernLayout`](https://github.com/scalar/scalar/blob/main/packages/api-reference/src/components/Layouts/ModernLayout.vue#L41) or [`ClassicLayout`](https://github.com/scalar/scalar/blob/main/packages/api-reference/src/components/Layouts/ClassicLayout.vue#L16).

The caveat would be you need to do it in a Vue app.

**Reply by hanspagel** [reply]
> Ability to hide scalar promo text

🥲

**Reply by kylemclaren** [reply]
@hanspagel Didn't mean to be insensitive to your efforts! Love Scalar, but would be nice to have some different options for the promo. Perhaps the ability to make it less prominent or move it to the footer.
